azure/services/scalesets/scalesets.go

```diff
@@ -424,13 +424,13 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 				NetworkProfile: &compute.VirtualMachineScaleSetNetworkProfile{
 					NetworkInterfaceConfigurations: &[]compute.VirtualMachineScaleSetNetworkConfiguration{
 						{
-							Name: to.StringPtr(vmssSpec.Name + "-netconfig"),
+							Name: to.StringPtr(vmssSpec.Name),
```

Comment:
is this a mutable property? what would happen for existing scale sets using this new code?

Reply:
We should be very confident this isn't a mutable property.

Interestingly, this change should be fine due to expected `NetworkProfile` mutations via cloud-provider-azure during the lifecycle of the VMSS:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/scalesets/scalesets.go#L720